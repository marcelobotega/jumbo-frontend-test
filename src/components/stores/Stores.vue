<template>
  <div>
    <h1>Stores</h1>

    <button @click.prevent="show = 'stores'" class="btn btn-blue mr-8">
      Show stores
    </button>
    <button @click.prevent="show = 'cities'" class="btn btn-blue">
      Show cities
    </button>
    <ul v-if="show === 'stores'">
      <li v-for="store in stores" :key="store.uuid">{{ store.addressName }}</li>
    </ul>
    <ul v-if="show === 'cities'">
      <li v-for="city in cities" :key="city">{{ city }}</li>
    </ul>
  </div>
</template>

<script lang="ts">
import stores from "../../data/stores-data";
import { Component, Vue } from "vue-property-decorator";
import { IStore } from "@/domain/store";

@Component
export default class Stores extends Vue {
  show = "";
  cities = ["Amsterdam", "Veghel"];
  stores: IStore[] = stores;
}
</script>

<style lang="scss" scoped>
ul {
  list-style: none;
}

li {
  background: #fdc513;
  display: inline-block;
  margin: 5px;
}
</style>
