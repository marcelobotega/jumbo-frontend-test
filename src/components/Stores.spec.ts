import { shallowMount } from "@vue/test-utils";
import Stores from "@/components/Stores.vue";

describe("Stores.vue", () => {
  it("should create", () => {
    const wrapper = shallowMount(Stores);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
