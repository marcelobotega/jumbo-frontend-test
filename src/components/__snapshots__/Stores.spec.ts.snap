// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stores.vue should create 1`] = `
<div><button class="btn btn-blue mr-8">
    Show stores
  </button> <button class="btn btn-blue">
    Show cities
  </button>
  <!---->
  <!---->
</div>
`;
